src/python/pants/option/options.py

```diff
@@ -285,6 +324,65 @@ def walk_parsers(self, callback):
     self._assert_not_frozen()
     self._parser_hierarchy.walk(callback)
 
+  def _check_deprecated_scope(self, scope, values):
+    # If we're the new name of a deprecated scope, also get values from that scope.
+    deprecated_scope = self.known_scope_to_info[scope].deprecated_scope
+    # Note that deprecated_scope and scope share the same Optionable class, so deprecated_scope's
+    # Optionable has a deprecated_options_scope equal to deprecated_scope. Therefore we must
+    # check that scope != deprecated_scope to prevent infinite recursion.
+    if deprecated_scope is not None and scope != deprecated_scope:
+      # Do the deprecation check only on keys that were explicitly set on the deprecated scope
+      # (and not on its enclosing scopes).
+      explicit_keys = self.for_scope(deprecated_scope,
+                                     inherit_from_enclosing_scope=False).get_explicit_keys()
+      if explicit_keys:
+        # Update our values with those of the deprecated scope (now including values inherited
+        # from its enclosing scope).
+        # Note that a deprecated val will take precedence over a val of equal rank.
+        # This makes the code a bit neater.
+        values.update(self.for_scope(deprecated_scope))
+
+        return dict(
+          removal_version=self.known_scope_to_info[scope].deprecated_scope_removal_version,
+          deprecated_entity_description='scope {}'.format(deprecated_scope),
+          hint='Use scope {} instead (options: {})'.format(scope, ', '.join(explicit_keys)))
+
+  _quiet_flag_regex = re.compile(r'\A(\-q|\-\-([^=]+\-)?quiet(=.*)?)\Z')
+
+  def _match_recursive_quiet_flag(self, scope, flags):
```

Comment:
Would be good to note here that the entire method can be removed after the deprecation.

Reply:
Also added in fb95e63972ee356bc3d1a7c23b361a88f2dae720!